crates/solidity/outputs/cargo/crate/src/generated/language.rs

```diff
@@ -0,0 +1,9977 @@
+                            break;
+                        }
+                        break;
+                    }
+                    helper.result(stream)
+                }) {
+                    break;
+                }
+                break;
+            }
+            helper.result()
+        }
+        .with_kind(RuleKind::ModifierDefinition)
+    }
+
+    #[allow(unused_assignments, unused_parens)]
+    pub fn modifier_invocation(&self, stream: &mut Stream) -> ParserResult {
+        {
+            let mut helper = SequenceHelper::new();
+            loop {
+                if helper.handle_next_result(self.identifier_path(stream)) {
+                    break;
+                }
+                if helper.handle_next_result(OptionalHelper::transform(
+                    self.arguments_declaration(stream),
+                )) {
+                    break;
+                }
+                break;
+            }
+            helper.result()
+        }
+        .with_kind(RuleKind::ModifierInvocation)
+    }
+
+    #[allow(unused_assignments, unused_parens)]
+    pub fn mul_div_mod_operator(&self, stream: &mut Stream) -> ParserResult {
+        {
+            let mut helper = ChoiceHelper::new(stream);
+            loop {
+                let result = self.default_parse_token_with_trivia(stream, TokenKind::Asterisk);
+                if helper.handle_next_result(stream, result) {
+                    break;
+                }
+                let result = self.default_parse_token_with_trivia(stream, TokenKind::Slash);
+                if helper.handle_next_result(stream, result) {
+                    break;
+                }
+                let result = self.default_parse_token_with_trivia(stream, TokenKind::Percent);
+                if helper.handle_next_result(stream, result) {
+                    break;
+                }
+                break;
+            }
+            helper.result(stream)
+        }
+    }
+
+    #[allow(unused_assignments, unused_parens)]
+    pub fn named_argument(&self, stream: &mut Stream) -> ParserResult {
+        {
+            let mut helper = SequenceHelper::new();
+            loop {
+                if helper.handle_next_result(
+                    self.default_parse_token_with_trivia(stream, TokenKind::Identifier),
+                ) {
+                    break;
+                }
+                if helper.handle_next_result(
+                    self.default_parse_token_with_trivia(stream, TokenKind::Colon),
+                ) {
+                    break;
+                }
+                if helper.handle_next_result(self.expression(stream)) {
+                    break;
+                }
+                break;
+            }
+            helper.result()
+        }
+        .with_kind(RuleKind::NamedArgument)
+    }
+
+    #[allow(unused_assignments, unused_parens)]
+    pub fn named_arguments_declaration(&self, stream: &mut Stream) -> ParserResult {
```

Comment:
Similar to how we expose a single entry point `parse(kind)` wrapper instead of individual `parse_rule()` methods, I suggest exposing a `scan(scope)` wrapper as well, instead of individual scanner methods, and for the same reasons.

Reply:
Additionally, I see the current exposed scanners accept a `&mut stream`. Is this intended? shouldn't we be accepting a string instead, similar to `parse(kind)`?
Stream is an internal type IIUC.

Reply:
done